source/distributed/fully_distributed_tria.cc

```diff
@@ -285,6 +289,74 @@ namespace parallel
 
 
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::copy_triangulation(
+      const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      // pointer to the triangulation for which the construction data
+      // should be created (normally it is the input triangulation but
+      // in the case of a serial triangulation we create a copy which should
+      // be used)
+      const dealii::Triangulation<dim, spacedim> *other_tria_ptr = &other_tria;
+
+      // temporary serial triangulation (since the input triangulation is const
+      // and we might modify its subdomain_ids
+      // and level_subdomain_ids during partitioning); it is filled only
+      // if the source triangulation is serial
+      std::unique_ptr<dealii::Triangulation<dim, spacedim>> serial_tria;
+
+      // check if other triangulation is not a parallel one, which needs to be
+      // partitioned
+      if (dynamic_cast<const dealii::parallel::TriangulationBase<dim, spacedim>
+                         *>(&other_tria) == nullptr)
+        {
+          serial_tria =
+            std::make_unique<dealii::Triangulation<dim, spacedim>>();
+
+          // actually copy the serial triangulation
+          serial_tria->copy_triangulation(
+            *const_cast<dealii::Triangulation<dim, spacedim> *>(&other_tria));
```

Comment:
Why do you need the `const_cast`? The `copy_triangulation` function takes a `const` reference.